mne/viz/evoked.py

```diff
@@ -2279,110 +2216,77 @@ def click_func(
-    # least color and linestyles.
-
-    ci_fun = _get_ci_function_for_evokeds(ci)
-
-    # if we have a dict/list of lists, we compute the grand average and the CI
+    # for each axis, compute the grand average and (maybe) the CI
     # (per sensor if topo, otherwise aggregating over sensors)
-    if not do_topo:
-        axes = [(ax, 0) for ax in axes]
-        if not (isinstance(picks, Iterable) and
-                isinstance(picks[0], Iterable)):
-            picks = [picks]  # because we iterate over it
-    else:
-        picks = list(picks)
-
-    all_data, all_cis = [], []
-    any_positive, any_negative = False, False
-    for picks_, (ax, idx) in zip(picks, axes):
-        data_dict, ci_dict = _calculate_ci_and_mean(
-            evokeds, conditions, scaling, picks_, ci_fun, gfp)
-        # we now have dicts for data ('evokeds' - grand averaged Evoked's)
-        # and the CI ('ci_array') with cond name labels
-        all_data.append(data_dict)
+    c_func = None if do_topo else combine_func
+    all_data = list()
+    all_cis = list()
+    for _picks, (ax, idx) in zip(picks, axes):
+        data_dict = dict()
+        ci_dict = dict()
+        for cond in conditions:
+            this_evokeds = evokeds[cond]
+            # skip CIs when possible; assign ci_fun first to get arg checking
+            ci_fun = _get_ci_function_pce(ci)
+            ci_fun = ci_fun if len(this_evokeds) > 1 else None
+            res = _get_data_and_ci(this_evokeds, combine, c_func, picks=_picks,
+                                   scaling=scalings, ci_fun=ci_fun)
+            data_dict[cond] = res[0]
+            if ci_fun is not None:
+                ci_dict[cond] = res[1]
+        all_data.append(data_dict)  # grand means, or indiv. sensors if do_topo
         all_cis.append(ci_dict)
     del evokeds
 
-    if do_topo:
-        picks.append(-1)  # so the loops don't terminate prematurely
+    # compute ylims
+    allvalues = list()
+    for _dict in all_data:
+        for _array in list(_dict.values()):
+            allvalues.append(_array[np.newaxis])  # to get same .ndim as CIs
+    for _dict in all_cis:
+        allvalues.extend(list(_dict.values()))
+    allvalues = np.concatenate(allvalues)
+    norm = np.all(allvalues > 0)
+    orig_ymin, orig_ymax = ylim.get(ch_type, [None, None])
+    ymin, ymax = _setup_vmin_vmax(allvalues, orig_ymin, orig_ymax, norm)
+    del allvalues
 
     # add empty data (all zeros) for the legend axis
-    all_data.append({cond: np.zeros(d.shape) for cond, d in data_dict.items()})
-    all_cis.append({cond: np.zeros(np.array(d).shape)
-                    for cond, d in ci_dict.items()})
+    all_data.append({cond: np.zeros(dat.shape)
+                     for cond, dat in data_dict.items()})
+    all_cis.append({cond: np.zeros(np.array(dat).shape)
+                    for cond, dat in ci_dict.items()})
+    if do_topo:
+        picks.append(-1)  # so we don't terminate loop before legend axis drawn
```

Comment:
Maybe we can use `last_iter` instead of -1.

Reply:
are you suggesting using a string instead of int, just to be more transparent?  +1 if that's what you mean.

Reply:
Indeed.

Reply:
turns out all we need is empty data, empty CI, and empty title to get this to work. appending to `picks` isn't even necessary any more.